File: cloud/oauth-server/src/main/java/pers/acp/admin/oauth/repo/MenuRepository.java
package pers.acp.admin.oauth.repo;

import pers.acp.admin.oauth.base.OauthBaseRepository;
import pers.acp.admin.oauth.entity.Menu;

import java.util.List;

/**
 * @author zhangbin by 2018-1-17 17:46
 * @since JDK 11
 */
public interface MenuRepository extends OauthBaseRepository<Menu, String> {

    List<Menu> findByAppid(String appId);

    void deleteByIdInAndCovert(List<String> idList, boolean covert);

}

File: cloud/oauth-server/src/main/kotlin/pers/acp/admin/oauth/repo/MenuRepository.kt
package pers.acp.admin.oauth.repo

import pers.acp.admin.common.base.BaseRepository
import pers.acp.admin.oauth.entity.Menu

/**
 * @author zhangbin by 2018-1-17 17:46
 * @since JDK 11
 */
interface MenuRepository : BaseRepository<Menu, String> {

    fun findAllByOrderBySortAsc(): MutableList<Menu>

    fun findByAppId(appId: String): MutableList<Menu>

    fun findByParentIdIn(idList: MutableList<String>): MutableList<Menu>

    fun deleteByIdInAndCovert(idList: MutableList<String>, covert: Boolean)

}
